node/state/core.go

```diff
@@ -31,9 +33,25 @@ func CoreAccessor(endpoint string) func(fx.Lifecycle, keystore.Keystore, params.
 			return nil, err
 		}
 		signer := apptypes.NewKeyringSigner(ring, keyringAccName, string(net))
+		keys, err := signer.List()
+		if err != nil {
+			return nil, err
+		}
+		// if no key was found in keystore path, generate new key for node
+		if len(keys) == 0 {
+			log.Infow("NO KEY FOUND IN PATH, GENERATING NEW KEY...", "path", ks.Path())
+			info, mn, err := signer.NewMnemonic(keyringAccName, keyring.English, "", "",
+				hd.Secp256k1)
+			if err != nil {
+				return nil, err
+			}
+			log.Info("NEW KEY GENERATED...")
+			fmt.Printf("\nNAME: %s\nADDRESS: %s\nMNEMONIC (save this somewhere safe!!!): \n%s\n\n",
```

Comment:
Should we print mnemonic? It is basically printing a private key to stdout or even worse, as its multiple private keys.

Reply:
Key utility prints it anyway

Reply:
It's a fresh-unfunded account.

Reply:
What's the alternative here?

Reply:
Not printing it? Writing to some file? TBH not sure, maybe it fine for now

Reply:
Writing it to an unencrypted file is probably less secure that printing it out on screen unencrypted. Writing to an encrypted file isn't a bad idea. Either way, can be left to later.

Reply:
I think in the SDK the user can chose if the mnemonic is printed or not.

Reply:
For now, we will log the mnemonic. If people want higher security for testnet, they can load their own keys into the path.

Reply:
Can I resolve @liamsi @adlerjohn

Reply:
Fine for now

Reply:
we have an issue that tracks keyring issues https://github.com/celestiaorg/celestia-node/issues/415